### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by madalinazanficu**
Created a button component, a wrapper on top of the Tamagui button, which extends TamaguiButtonProps and supports 4 different presets: default, outlined, chromeless, and red.

image_123650291

### CODE REVIEW

File: mobile/components/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    return (
+      <TamaguiButton {...rest} style={$styles}>
+        {children}
+      </TamaguiButton>
+    );
+}
+
+
+const $presets = (theme: any, tokens: any) => {
```
[the remark this answers is not in the document]
Comment (madalinazanficu) [reply mobile/components/Button.tsx]: Done, as disscussed with Lucia.

File: mobile/components/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    const preset = $presets(theme, tokens)[presetType]
+
+    const $styles: StyleProp<ViewStyle> = [
+      preset,
+      $styleOverride,
+    ];
+
+    return (
+      <TamaguiButton {...rest} style={$styles}>
```
[the remark this answers is not in the document]
Comment (madalinazanficu) [reply mobile/components/Button.tsx]: Done. 
Button components supports now: text and textStyle props and incorporates Typography with the preset specific color, fontSize of 16px and fontWeight of 500.

File: mobile/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  };
+
+  return {
+      default: $baseStyle, 
+
+      outlined: {
+          ...$baseStyle,
+          borderWidth: 1,
+          borderColor: theme.$purple5?.val,
+          backgroundColor: 'transparent',
```
Comment (luciatugui) [inline mobile/components/Button.tsx:92]: this should be white as per design

**Comment by luciatugui** [comment]
As per our discussion on slack, the Button component should also handle the pressed state, with a darkening of the background.
The darker colours should also be defined inside the theme.

**Review by birloiflorian — APPROVED** [review]

**Pull request merged**